### DISCUSSION TITLE
"subscriptionOffers are required for Google Play Subscriptions"

### DISCUSSION THREAD

**Discussion by omidbarcelona** [hyochan/react-native-iap#2029] [General]
Hi,
Using I get RNIap.requestSubscription(), I get this error "subscriptionOffers are required for Google Play Subscriptions".
RNIap.getSubscription() works correctly.

See the code below. Any solution?
```

RNIap.requestSubscription(itemSku).then(
                    result => {
                        console.log('result', JSON.stringify(result))
                    }
                ).catch(e => console.log('requestSubscription error: ', e))
```

**Comment by andresesfm** [comment]
Read migration to 10.0.0 in the docs

**Reply by omidbarcelona** [reply]
I follow migration 10.0.0

```
getProducts({skus: ['my_sku']});
import * as IAP from 'react-native-iap'
```

as I said in the post, RNIap.getSubscription() works correctly, but RNIap.requestSubscription() does not work.

**Reply by omidbarcelona** [reply]
problem solve with following code
```
const subscriptionOffers = [{
            sku,
            offerToken,
        }]
        RNIap.requestSubscription({ subscriptionOffers }).then((result) => {
            console.log("result: " + JSON.stringify(result))
        }).catch(e => console.log('requestSubscription error: ', e))
```

**Reply by thuongtv-vn** [reply]
Hi @omidbarcelona 
I having a problem when requestSubscription success. It is returning a boolean value instead of an object `SubscriptionPurchase` that contain transaction info. In order to get transaction info, I must call `getAvailablePurchases`.

is it true that the api return a boolean value? I tested on Android device.

**Reply by andresesfm** [reply]
@thuongtv-vn this has been fixed on `12.3.2`
